### PR TITLE
feat(case-converter): add text case converter and slug generator (#76)

### LINKED ISSUE #76: feat: Text Case Converter & Slug Generator — camelCase, snake_case, kebab-case, and more
## Summary
Add a **Text Case Converter & Slug Generator** tool. Developers convert between casing conventions dozens of times a day — renaming DB columns to match TypeScript fields, generating CSS classes from camelCase, creating URL slugs from titles. Currently scattered across ad-laden websites that log your input.

## Why Privacy Matters Here
Developers paste proprietary variable names, internal product codenames, microservice endpoint names, and database schema identifiers into random web converters that track sessions and log inputs. **Toolglass does this 100% client-side.**

## Proposed Implementation
- **Location:** `src/tools/case-converter/CaseConverterTool.tsx`
- **Zero dependencies** — pure JS string manipulation
- **Supported conversions:**
  - `camelCase` ↔ `PascalCase`
  - `snake_case` ↔ `CONSTANT_CASE`
  - `kebab-case` ↔ `Title Case`
  - `dot.notation` ↔ `path/case`
  - `Sentence case` ↔ `lowercase` / `UPPERCASE`
  - **URL Slug** — strips diacritics (`Café` → `cafe`), replaces spaces with hyphens, removes special chars
- **Features:**
  - Paste text → see all conversions simultaneously in a live grid
  - Copy any individual conversion
  - Batch mode: paste multiple lines, convert all at once
  - Auto-detect input casing
  - Uses `String.prototype.normalize("NFD")` for proper unicode handling

## Acceptance Criteria
- [ ] All listed case conversions work correctly
- [ ] Unicode/diacritics handled properly (é → e, ñ → n)
- [ ] Batch multi-line mode works
- [ ] Copy individual results
- [ ] Registry entry

### PR DESCRIPTION
**Pull request by ajithakdev**
## Summary
Closes #76

Adds a comprehensive, zero-dependency, 100% client-side **Text Case Converter & Slug Generator** tool:
- **12 Supported Conversions**:
  - `camelCase`
  - `PascalCase`
  - `snake_case`
  - `CONSTANT_CASE`
  - `kebab-case`
  - `URL Slug` (with diacritic decomposition & normalization, punctuation stripping, trim dashes)
  - `Title Case`
  - `Sentence case`
  - `dot.case`
  - `path/case`
  - `lowercase`
  - `UPPERCASE`
- **Automatic Casing Detection**: Recognizes current casing format and informs the user in real-time.
- **Batch Multi-Line Transformation**: Convert lists/lines of variables or text to any target format with one click.
- **Live Output Grid**: Responsive cards with individual 1-click copy buttons and toast feedback.
- **Quick Preset Samples**: 5 sample inputs demonstrating different casing structures.
- **URL Synchronization**: Shareable state with `useUrlState` for instant linking.
- **Registered & Categorized**: Added to registry with `Type` icon and categorized under 'Converters & Formatters' on the Landing page.

## Testing & Verification
- Unit test suite: `src/tools/case-converter/caseConverter.test.ts` (10 unit tests) covering all 12 casing transformations, Unicode diacritics stripping, edge cases, punctuation handling, and casing detection.
- Component test suite: `src/tools/case-converter/caseConverterTool.test.tsx` (5 component tests) covering layout, real-time input updating, sample loading, batch conversion, and casing badge.
- Full project test suite passed: `56 passed across 9 test files`
- Full typecheck (`tsc -b`), lint (`eslint .`), and production bundle build (`vite build`) verified with zero errors.

---

<a href="https://app.devin.ai/review/ajithakdev/toolglass/pull/94" target="_blank">
  <picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://static.devin.ai/assets/gh-devin-review-dark.svg?v=3">
    Devin Review
  </picture>
</a>

### CODE REVIEW

File: src/tools/case-converter/caseConverter.ts [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+  if (!input || !input.trim()) return [];
+
+  // 1. Separate camelCase/PascalCase boundaries (e.g. "fooBar" -> "foo Bar", "XMLHttpRequest" -> "XML Http Request")
+  const separated = input
+    .replace(/([a-z0-9])([A-Z])/g, '$1 $2')
+    .replace(/([A-Z]+)([A-Z][a-z])/g, '$1 $2');
+
+  // 2. Split by any non-alphanumeric separator (spaces, underscores, hyphens, dots, slashes)
+  const matches = separated.match(/[a-zA-Z0-9]+/g);
+  return matches || [];
```
[the remark this answers is not in the document]
Comment (ajithakdev) [reply src/tools/case-converter/caseConverter.ts]: Resolved in commit b12f45e. Updated `extractWords`, `detectCasing`, and `toSlug` to use Unicode property escapes (`\p{Ll}`, `\p{Lu}`, `\p{Lt}`, `\p{L}`, `\p{N}`), preserving accented characters (e.g. `caféLatte` -> `café_latte`) and supporting non-Latin scripts (Cyrillic, Greek, etc.) across all 12 casing conversions and slug generation with accompanying unit tests.

File: src/tools/case-converter/caseConverter.ts [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+  if (!trimmed) return 'Unknown';
+
+  if (/^[\p{Ll}]+(?:[\p{Lu}\p{Lt}][\p{Ll}\p{N}]*)+$/u.test(trimmed)) return 'camelCase';
+  if (/^[\p{Lu}\p{Lt}][\p{Ll}\p{N}]+(?:[\p{Lu}\p{Lt}][\p{Ll}\p{N}]*)+$/u.test(trimmed)) return 'PascalCase';
+  if (/^[\p{Ll}\p{N}]+(?:_[\p{Ll}\p{N}]+)+$/u.test(trimmed)) return 'snake_case';
+  if (/^[\p{Lu}\p{Lt}\p{N}]+(?:_[\p{Lu}\p{Lt}\p{N}]+)+$/u.test(trimmed)) return 'CONSTANT_CASE';
+  if (/^[\p{Ll}\p{N}]+(?:-[\p{Ll}\p{N}]+)+$/u.test(trimmed)) return 'kebab-case';
+  if (/^[\p{Ll}\p{N}]+(?:\.[\p{Ll}\p{N}]+)+$/u.test(trimmed)) return 'dot.case';
+  if (/^[\p{Ll}\p{N}]+(?:\/[\p{Ll}\p{N}]+)+$/u.test(trimmed)) return 'path/case';
+
```
[the remark this answers is not in the document]
Comment (ajithakdev) [reply src/tools/case-converter/caseConverter.ts:104]: Resolved in commit 7a6b057. Added explicit single-word casing detection checks for `lowercase`, `UPPERCASE`, and `Capitalized` words in `detectCasing`, while keeping multi-word / sentence strings classified as `Natural text`.
